catalog/app/utils/json-schema.js

```diff
@@ -1,82 +1,96 @@
 import * as R from 'ramda'
 import isArray from 'lodash/isArray'
+import isBoolean from 'lodash/isBoolean'
+import isNull from 'lodash/isNull'
 import isNumber from 'lodash/isNumber'
 import isObject from 'lodash/isObject'
 import isString from 'lodash/isString'
-import isUndefined from 'lodash/isUndefined'
 
-function isSchemaArray(optSchema) {
-  return R.prop('type', optSchema) === 'array'
-}
+const isSchemaArray = (optSchema) => R.prop('type', optSchema) === 'array'
 
-function isSchemaObject(optSchema) {
-  return R.prop('type', optSchema) === 'object'
-}
+const isSchemaObject = (optSchema) => R.prop('type', optSchema) === 'object'
 
-function isSchemaString(optSchema) {
-  return R.prop('type', optSchema) === 'string' && !isSchemaEnum(optSchema)
-}
+const isSchemaString = (optSchema) => R.prop('type', optSchema) === 'string'
 
-function isSchemaNumber(optSchema) {
-  return R.prop('type', optSchema) === 'number'
-}
+const isSchemaNumber = (optSchema) => R.prop('type', optSchema) === 'number'
 
-function isSchemaBoolean(optSchema) {
-  return R.prop('type', optSchema) === 'boolean'
-}
+const isSchemaInteger = (optSchema) => R.prop('type', optSchema) === 'integer'
 
-function isSchemaNull(optSchema) {
-  return R.prop('type', optSchema) === 'null'
-}
+const isSchemaBoolean = (optSchema) => R.prop('type', optSchema) === 'boolean'
 
-export function isSchemaEnum(optSchema) {
-  return !!R.prop('enum', optSchema)
-}
+const isSchemaNull = (optSchema) => R.prop('type', optSchema) === 'null'
 
-function isSchemaConst(optSchema) {
-  return !!R.prop('const', optSchema)
-}
+export const isSchemaEnum = (optSchema) => !!R.prop('enum', optSchema)
+
+export const isSchemaOneOf = (optSchema) => !!R.prop('oneOf', optSchema)
+
+export const isSchemaAnyOf = (optSchema) => !!R.prop('anyOf', optSchema)
+
+export const isSchemaAllOf = (optSchema) => !!R.prop('allOf', optSchema)
+
+const isSchemaConst = (optSchema) => !!R.prop('const', optSchema)
 
 function isSchemaCompound(optSchema) {
   if (!optSchema) return false
   return ['anyOf', 'oneOf', 'not', 'allOf'].some((key) => optSchema[key])
 }
 
-function isSchemaReference(optSchema) {
-  return !!R.prop('$ref', optSchema)
-}
+const isSchemaReference = (optSchema) => !!R.prop('$ref', optSchema)
 
 export const isNestedType = R.either(isSchemaArray, isSchemaObject)
 
+const compoundTypeToHumanString = (optSchema, condition, divider) =>
+  R.prop(condition, optSchema)
```

Comment:
i think this can throw an error, bc `R.prop` can return undefined if `optSchema` is undefined (or is it always defined?). also, you could just use the vanilla js `optSchema[condition]` here, bc `R.prop` benefits (currying and not throwing on undefined) dont seem to be required here.

Reply:
It's always defined, but I should expect undefined too. WIl fix it